src/main/java/io/openliberty/tools/intellij/lsp4mp/lsp4ij/server/ProcessStreamConnectionProvider.java

```diff
@@ -160,5 +169,44 @@ public String toString() {
         return "ProcessStreamConnectionProvider [commands=" + this.getCommands() + ", workingDir=" //$NON-NLS-1$//$NON-NLS-2$
                 + this.getWorkingDirectory() + "]"; //$NON-NLS-1$
     }
+    public boolean isJavaHomeValid(String javaHome, String serverType) {
+
+        if (javaHome == null) {
+            String errorMessage = LocalizedResourceUtil.getMessage("javaHome.is.null", serverType, Constants.REQUIRED_JAVA_VERSION);
+            LOGGER.error(errorMessage);
+            showErrorPopup(errorMessage);
+            return false;
+        }
+
+        File javaHomeDir = new File(javaHome);
+        if (!javaHomeDir.exists()) {
+            String errorMessage = LocalizedResourceUtil.getMessage("javaHomeDir.does.not.exist", serverType, Constants.REQUIRED_JAVA_VERSION);
+            LOGGER.error(errorMessage);
+            showErrorPopup(errorMessage);
+            return false;
+        }
+
+        if (!checkJavaVersion(javaHome, Constants.REQUIRED_JAVA_VERSION)) {
+            String errorMessage = LocalizedResourceUtil.getMessage("java.version.message", serverType, Constants.REQUIRED_JAVA_VERSION);
+            LOGGER.error(errorMessage);
+            showErrorPopup(errorMessage);
+            return false;
+        }
+        return true;
+    }
+
+    private void notifyError(String errMsg, Project project) {
+        Notification notif = new Notification(Constants.LIBERTY_DEV_DASHBOARD_ID, errMsg, NotificationType.WARNING)
+                .setTitle(LocalizedResourceUtil.getMessage("java.runtime.error.message"))
+                .setIcon(AllIcons.General.Warning)
```

Comment:
Let's use the Liberty icon here to ensure its clear the message is coming from the Liberty Tools plugin